### ISSUE TITLE
Can vale check for empty headers and links in markdown?

### ISSUE THREAD

**Issue by tperry-r7** [vale-cli/vale#193]
- OS macOs
- Install method - Homebew
- Vale version vale version 2.1.0

I have been trying to configure two existence rules for empty links and empty headings. Both of these work in a regex tester, but vale is not pulling it in. Looking for empty items is the only think it seems to be struggling with. Thanks. 

1. Check for empty headings. 
https://regex101.com/r/c4Xsgj/1/
```yaml
extends: existence
message: Consider removing empty headings '%s'
ignorecase: true
scope: heading
level: warning
token: '^#+ *$'

# (<=\s|^)###(.*?)$
# ^#+ *$
```

2. Check for empty links
```yaml
extends: existence
message: "Don't use empty links '%s'"
ignorecase: true
level: warning
scope: link
nonword: true
tokens:
    - (?:\[\]\(\))
    - (?:\W\[\]\(\))
    - (?:\[\]\(\))
    - '(?:\(\))'
```

**Comment by jdkato** [comment]
You'll want to use [`scope: raw`](https://errata-ai.gitbook.io/vale/getting-started/markup#available-scopes) (available since v2.1.0) whenever you're trying to match markup syntax (e.g., `#` or `[...]`) itself.

**Comment by kayx23** [comment]
> You'll want to use [`scope: raw`](https://errata-ai.gitbook.io/vale/getting-started/markup#available-scopes) (available since v2.1.0) whenever you're trying to match markup syntax (e.g., `#` or `[...]`) itself.

updated link: https://vale.sh/docs/topics/scoping/#markup
